src/main/kotlin/org/wa/rceditor/application/model/MainModel.kt

```diff
@@ -1,26 +1,310 @@
+        checkingEntities.add(CheckingEntityItem(text))
+    }
+
+    fun removeCheckingEntity(item: CheckingEntityItem) {
+        checkingEntities.remove(item)
+    }
+
+    fun projects(): ObservableList<ProjectItem> {
+        return projects
+    }
+
+    fun addProject(title: String, versification: String, identifier: String, sort: Int, path: String, category: String) {
+        projects.add(ProjectItem(title, versification, identifier, sort, path, category))
+    }
+
+    fun removeProject(item: ProjectItem) {
+        projects.remove(item)
+    }
+
+    // ---------- Handlers --------- //
+
     fun onNewDocumentSelected() {
-        println("New document created")
+        chooseDirectory("Create Resource Container").apply {
+            if (this != null) {
+                directory = this
+                container = ResourceContainer.create(directory) {
+                    this.manifest = Manifest(DublinCore(), listOf(), Checking())
+                }
+                clearData()
+                directoryLoaded = true
+            }
+        }
     }
 
     fun onOpenDocumentSelected() {
-        println("Document Opened")
+        chooseDirectory("Open Resource Container").apply {
+            if (this != null) {
+                directory = this
+                container = ResourceContainer.load(directory)
+                clearData()
+                directoryLoaded = true
+                loadRecourceContainer()
+            }
+        }
     }
 
     fun onSaveDocumentSelected() {
-        println("Document Saved")
+        saveResourceContainer()
     }
 
     fun onAppQuitSelected() {
         exitProcess(-1)
     }
 
-    fun onEditContributorsClick() {
+    fun loadRecourceContainer() {
+        // TODO load on background thread
+
+        conformsto = container.manifest.dublinCore.conformsTo
+        creator = container.manifest.dublinCore.creator
+        description = container.manifest.dublinCore.description
+        format = container.manifest.dublinCore.format
+        identifier = container.manifest.dublinCore.identifier
+        issued = LocalDate.parse(container.manifest.dublinCore.issued, DateTimeFormatter.ISO_DATE)
+        modified = LocalDate.parse(container.manifest.dublinCore.modified, DateTimeFormatter.ISO_DATE)
+        languageDirection = container.manifest.dublinCore.language.direction
+        languageIdentifier = container.manifest.dublinCore.language.identifier
+        languageTitle = container.manifest.dublinCore.language.title
+        publisher = container.manifest.dublinCore.publisher
+        rights = container.manifest.dublinCore.rights
+        subject = container.manifest.dublinCore.subject
+        title = container.manifest.dublinCore.title
+        type = container.manifest.dublinCore.type
+        version = container.manifest.dublinCore.version
+        checkingLevel = container.manifest.checking.checkingLevel
+
+        container.manifest.dublinCore.contributor.forEach {
```

Comment:
Might be able to simplify to `contributors.addAll(container.manifest.dublinCore.contributor)`

Reply:
Similar for following lines in this file as well